### PR TITLE
SystemD updates

### PR DESCRIPTION
**Pull request by Graeme-Miller**
Updated docker ssh driver such that it will work correctly with SystemD based OSes

### CODE REVIEW

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 1]
```diff
@@ -34,7 +27,6 @@
 import com.google.common.base.Stopwatch;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Lists;
-
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Need to fix imports. I can provide configuration for Eclipse, not sure about IntelliJ

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 2]
```diff
@@ -321,19 +313,51 @@ public void customize() {
...
+                .failOnNonZeroResultCode()
+                .execute();
+
+
+        //CentOS config
+        newScript(CUSTOMIZING)
+                .body.append(
+                chain(
+                        sudo("mkdir -p /etc/sysconfig"),
+                format("echo 'other_args=\"--selinux-enabled -H tcp://0.0.0.0:%d -H unix:///var/run/docker.sock -e lxc %s --tlsverify --tls --tlscert=%s/cert.pem --tlskey=%<s/key.pem --tlscacert=%<s/ca.pem\"' | ",
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Indent

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 3]
```diff
@@ -321,19 +313,51 @@ public void customize() {
...
+                                getDockerPort(), getStorageOpts(), getRunDir()) + sudo("tee -a /etc/sysconfig/docker")
+                ))
+                .failOnNonZeroResultCode()
+                .execute();
+
+        //SystemD
+        String systemDConfigInstallPath = Os.mergePaths(getInstallDir(), "docker.service");
+
+
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Extra blank lines need removed

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 4]
```diff
@@ -321,19 +313,51 @@ public void customize() {
...
-                                sudo(format("gpasswd -a %s docker", getMachine().getUser())),
-                                sudo("newgrp docker"))),
-                        ifExecutableElse0("yum",
-                                format("echo 'other_args=\"--selinux-enabled -H tcp://0.0.0.0:%d -H unix:///var/run/docker.sock -e lxc %s --tlsverify --tls --tlscert=%s/cert.pem --tlskey=%<s/key.pem --tlscacert=%<s/ca.pem\"' | ",
-                                                getDockerPort(), getStorageOpts(), getRunDir()) +
-                                        sudo("tee -a /etc/sysconfig/docker")))
+                chain(
+                        sudo("mkdir -p /etc/systemd/system"),
+                        sudo(String.format("cp %s %s", systemDConfigInstallPath, "/etc/systemd/system/docker.service")),
+                        ifExecutableElse0("systemctl", sudo("systemctl daemon-reload"))
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Only do the chain of `mkdir`, `cp` and `systemctl` if `systemctl` exists.

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 5]
```diff
@@ -17,14 +17,7 @@
...
 import static java.lang.String.format;
-import static org.apache.brooklyn.util.ssh.BashCommands.INSTALL_CURL;
-import static org.apache.brooklyn.util.ssh.BashCommands.alternatives;
-import static org.apache.brooklyn.util.ssh.BashCommands.chainGroup;
-import static org.apache.brooklyn.util.ssh.BashCommands.fail;
-import static org.apache.brooklyn.util.ssh.BashCommands.ifExecutableElse0;
-import static org.apache.brooklyn.util.ssh.BashCommands.ifExecutableElse1;
-import static org.apache.brooklyn.util.ssh.BashCommands.installPackage;
-import static org.apache.brooklyn.util.ssh.BashCommands.sudo;
+import static org.apache.brooklyn.util.ssh.BashCommands.*;
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Prefer explicit imports to wildcard.

**Comment by grkvlt** [comment]
Mostly formatting, but also change the `systemd` customisation script to only execute if required, and set different names for the scripts.

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 6]
```diff
@@ -321,19 +313,51 @@ public void customize() {
                     .execute();
         }
 
+        //Create config for upstart
+        newScript(CUSTOMIZING)
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Give these scripts different names as `newScript(CUSTOMIZING + "-upstart")` and so on.

**Comment by grkvlt** [comment]
@Graeme-Miller Suggest you squash the commits once you've made the changes, too.

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java (hunk 2 above)
Comment (Graeme-Miller) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: too much or not enough? The InteliJ formatter seems to think this is correct

**Pull request merged**